src/uct/base/uct_iov.inl

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCT_IOV_INL_
+#define UCT_IOV_INL_
+
+#include <uct/api/uct.h>
+#include <ucs/sys/math.h>
+#include <ucs/debug/assert.h>
+
+/**
+ * Calculates total length of particular iov data buffer.
+ * Currently has no support for stride.
+ * If stride supported it should be like: length + ((count - 1) * stride)
+ */
+static UCS_F_ALWAYS_INLINE
+size_t uct_iov_get_length(const uct_iov_t *iov)
+{
+    return iov->count * iov->length;
+}
+
+/**
+ * Calculates total length of the iov array buffers.
+ */
+static UCS_F_ALWAYS_INLINE
+size_t uct_iov_total_length(const uct_iov_t *iov, size_t iovcnt)
+{
+    size_t iov_it, total_length = 0;
+
+    for (iov_it = 0; iov_it < iovcnt; ++iov_it) {
+        total_length += uct_iov_get_length(&iov[iov_it]);
+    }
+
+    return total_length;
+}
+
+/**
+ * Update an array of UCT IOV elements to consider an already consumed data.
+ *
+ * @param [in]     iov            A pointer to an array of UCT IOV elements.
+ * @param [in]     iov_cnt        A number of elements in a iov array.
+ * @param [in/out] cur_iov_idx    A pointer to an index in a iov array from
+ *                                which the operation should be started.
+ * @param [in]     consumed       An amount of data consumed that should be
+ *                                considered in a current iov array.
+ */
+static UCS_F_ALWAYS_INLINE
+void uct_iov_advance(uct_iov_t *iov, size_t iovcnt,
```

Comment:
what the the use case for this function?
the user should maintain own iov_index+iov_offset, not modify iovec.
does it mean user goes over iovec from beginning every time?